src/bp-activity/bp-activity-filters.php

```diff
@@ -2360,25 +2043,489 @@ function bp_blogs_activity_content_with_read_more( $content, $activity ) {
+						bb_migration_remove_activity_id_activity_update_type( $sub_activity_id_store, $get_parent_id->id );
+						bp_activity_update_meta( $get_parent_id->id, 'bp_media_comment_migration', 'success' );
+						/* translators: %s refers to how many numbers running at that process */
+						$records_updated = sprintf( __( '%s media comment migrated successfully.', 'buddyboss' ), number_format_i18n( $offset ) );
+					} else {
+						/* translators: %s refers to how many numbers running at that process */
+						$records_updated = sprintf( __( '%s no media comment available.', 'buddyboss' ), number_format_i18n( $offset ) );
+					}
+				}
+			} else {
+				/* translators: %s refers to how many numbers running at that process */
+				$records_updated = sprintf( __( '%s media already migrated.', 'buddyboss' ), number_format_i18n( $offset ) );
+			}
+			$offset ++;
+		}
+		bb_migration_write_log( 'offset - ' . $offset );
+		bb_migration_write_log( '$recipients_count_row_data->ids - ' . $recipients_count_row_data->ids );
+		if ( 1 === (int) $recipients_count_row_data->ids || (int) $offset === (int) $recipients_count_row_data->ids ) {
+			return array(
+				'status'  => 1,
+				'message' => __( 'media comment migration complete!', 'buddyboss' ),
+			);
+		} else {
+			return array(
+				'status'  => 'running',
+				'offset'  => $offset,
+				'records' => $records_updated,
+			);
+		}
+	} else {
+		return array(
+			'status'  => 1,
+			'message' => __( 'media comment migration complete!', 'buddyboss' ),
+		);
+	}
+}
+/**
+ * Get children data based on root element id. Its also get child data based on his parent id.
+ *
+ * @param array $activity_id_store store children data and check it should not call second time.
+ *
+ * @param array $sub_data          Get array of parent ids children data. Its work as herarchical array.
+ *
+ * @param array $children_array    Array of children data. Here all childrens check his own children.
+ *                                 If any children have its own child data then it will store in $sub_data array.
+ *
+ * @param int   $main_root_id      Main root element id.
+ *
+ * @return array|void Return array with child_array and activity_id_store
+ */
+function bb_migration_get_children_data( $activity_id_store, $sub_data, $children_array, $main_root_id ) {
+	global $wpdb;
+	$bp                   = buddypress();
+	$activity_table_name  = $bp->activity->table_name;
+	$child_array          = array();
+	$main_sub_child_array = array();
+	foreach ( $children_array as $children_data ) {
+		$activity_id_store = array_map( 'intval', $activity_id_store );
+		if ( in_array( (int) $children_data->id, $activity_id_store, true ) ) {
+			return;
+		}
+		$activity_id_store[] = $children_data->id;
+		$args                = array(
+			'display_comments' => true,
+			'show_hidden'      => true,
+			'sort'             => 'ASC',
+			'activity_ids'     => $children_data->id,
+		);
+		$get_activity_data   = bp_activity_get_specific( $args );
```

Comment:
here too, we don't need `get` in variable name, eg. `$get_activity_data` -> `$activity_data`